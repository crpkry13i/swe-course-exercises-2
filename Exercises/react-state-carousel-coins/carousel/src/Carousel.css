.Carousel-main {
  display: flex;
  align-items: center;
  justify-content: center;
}

.Carousel-main>i {
  cursor: pointer;
  margin: 10px;
  color: #aaa;
}

.Carousel-main>i.hidden {
  opacity: 0;
  pointer-events: none;
}

.Carousel-main>i:hover {
  color: black;
}