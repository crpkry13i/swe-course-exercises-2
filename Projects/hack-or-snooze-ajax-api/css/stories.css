/* Lists of stories */

.stories-list {
  margin: 20px 5px;
}

.stories-list > li {
  color: gray;
  font-size: 0.8rem;
  margin: 10px 0;
}

#stories-loading-msg {
  font-weight: bold;
  font-size: 150%;
  margin: 20px 30px;
}


/* Individual stories */

.story-link {
  color: black;
  font-size: 0.85rem;
  font-weight: normal;
  margin: 18px 0;
}

.story-link:hover {
  text-decoration: none;
}

.story-author {
  margin-left: 2em;
}

.story-user {
  display: block;
}
