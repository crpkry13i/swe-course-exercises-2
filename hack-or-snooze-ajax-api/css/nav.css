/* Navigation bar */

nav {
  display: flex;
  background-color: #ff6600;
  align-items: center;
  padding: 5px 20px;
  border-radius: 3px 3px 0 0;
}

.navbar-brand {
  font-weight: bold;
}

.nav-link {
  font-size: 0.85rem;
  margin: 0 3px;
}

.nav-right {
  margin-left: auto;
  text-align: right;
}
