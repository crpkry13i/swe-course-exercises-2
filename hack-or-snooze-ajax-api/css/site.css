/* General typography */

body {
  font-family: Arimo, sans-serif;
  margin: 8px 7.5vw;
}

h1 {
  font-size: 1.1rem;
  margin: 0;
}

h4 {
  font-size: 1rem;
  margin: 0;
}

h5 {
  font-size: 0.9rem;
  font-weight: lighter;
}

a {
  text-decoration: none;
  color: inherit;
}

a:hover {
  text-decoration: underline;
}


/* Site layout */

/* This is the basic box that the main part of the page goes into */
.container {
  display: flex;
  flex-direction: column;
  align-self: center;
  background-color: #f6f6ef;
}

.hidden {
  display: none;
}


/* Forms */

form {
  display: flex;
  flex-direction: column;
  margin: 8px 18px 0;
}

form > * {
  margin: 10px 0;
}

form label {
  font-size: 0.9rem;
  font-weight: 700;
  display: inline-block;
  width: 3.5rem;
  text-align: right;
  margin-right: 5px;
}

form input {
  font-size: 0.8rem;
  border: none;
  border-radius: 2px;
  padding: 8px;
  width: 300px;
  box-shadow: 0 0 3px 1px lightgray;
}

form input:focus {
  outline: none;
  box-shadow: 0 0 4px 1px darkgray;
}

form > button {
  width: 4rem;
  margin: 5px 0 15px 65px;
  border: none;
  border-radius: 4px;
  padding: 8px;
  font-size: 0.85rem;
  background-color: lightslategray;
  color: white;
  cursor: pointer;
  transition: all 0.15s;
}

form > button:hover {
  background-color: dimgray;
}

form > hr {
  margin: 0;
  border: 1px solid lightgray;
}

.login-input label {
  width: 70px;
}


/* responsive queries for tightening things up for mobile. */

@media screen and (max-width: 576px) {
  body {
    margin: 0;
  }
}

@media screen and (min-width: 992px) {
  body {
    max-width: 900px;
    margin: 8px auto;
  }
}
