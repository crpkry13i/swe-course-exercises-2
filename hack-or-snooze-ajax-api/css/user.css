/* Login and signup forms */

.account-form button {
  width: 4rem;
  margin-left: 80px;
}

#signup-form button {
  width: 8rem;
}

.account-forms-container {
  padding-left: 20px;
}
