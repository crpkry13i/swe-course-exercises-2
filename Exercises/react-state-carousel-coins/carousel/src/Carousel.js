import React, { useState } from "react";
import "./Carousel.css";
import image1 from "./image1.jpg";
import image2 from "./image2.jpg";
import image3 from "./image3.jpg";
import Card from "./Card";

function Carousel(props) {
  const [cardIdx, setCardIdx] = useState(0);
  const card = props.cardData[cardIdx];
  const total = props.cardData.length;
  const leftIconHidden = cardIdx === 0 ? "hidden" : "";
  const rightIconHidden = cardIdx === total - 1 ? "hidden" : "";
  const goForward = () => setCardIdx(cardIdx + 1);
  const goBackward = () => setCardIdx(cardIdx - 1);

  return (
    <div className="Carousel">
      <h1>{props.title}</h1>
      <div className="Carousel-main">
        <i
          className={`fas fa-chevron-circle-left fa-2x ${leftIconHidden}`}
          onClick={goBackward}
          data-testid="left-arrow"
        />
        <Card
          caption={card.caption}
          src={card.src}
          currNum={cardIdx + 1}
          totalNum={total}
        />
        <i
          className={`fas fa-chevron-circle-right fa-2x ${rightIconHidden}`}
          onClick={goForward}
          data-testid="right-arrow"
        />
      </div>
    </div>
  );
}

Carousel.defaultProps = {
  cardData: [
    {
      src: image1,
      caption: "Photo by Richard Pasquarella on Unsplash",
    },
    {
      src: image2,
      caption: "Photo by Pratik Patel on Unsplash",
    },
    {
      src: image3,
      caption: "Photo by Josh Post on Unsplash",
    },
  ],
  title: "Shells from far away beaches.",
};

export default Carousel;
