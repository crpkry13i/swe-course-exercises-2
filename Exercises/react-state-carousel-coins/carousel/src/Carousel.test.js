import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Carousel from "./Carousel";
import "./Carousel.css";

// smoke test
it("renders without crashing", () => {
  render(<Carousel />);
});

// snapshot test
it("matches snapshot", () => {
  const { asFragment } = render(<Carousel />);
  expect(asFragment()).toMatchSnapshot();
});

it("works when you click on the right arrow", function () {
  const { queryByTestId, queryByAltText } = render(<Carousel />);

  // expect the first image to show, but not the second
  expect(
    queryByAltText("Photo by Richard Pasquarella on Unsplash")
  ).toBeInTheDocument();
  expect(
    queryByAltText("Photo by Pratik Patel on Unsplash")
  ).not.toBeInTheDocument();

  // move forward in the carousel
  const rightArrow = queryByTestId("right-arrow");
  fireEvent.click(rightArrow);

  // expect the second image to show, but not the first
  expect(
    queryByAltText("Photo by Richard Pasquarella on Unsplash")
  ).not.toBeInTheDocument();
  expect(
    queryByAltText("Photo by Pratik Patel on Unsplash")
  ).toBeInTheDocument();
});

test("when on second image, clicking left arrow will move to the first image", () => {
  const { queryByTestId, getByText } = render(<Carousel />);
  const leftArrow = queryByTestId("left-arrow");
  const rightArrow = queryByTestId("right-arrow");

  fireEvent.click(rightArrow);

  expect(getByText("Photo by Pratik Patel on Unsplash")).toBeInTheDocument();

  fireEvent.click(leftArrow);

  expect(
    getByText("Photo by Richard Pasquarella on Unsplash")
  ).toBeInTheDocument();
});

it("hides and shows arrows appropriately", function () {
  const { getByTestId } = render(<Carousel />);
  const leftArrow = getByTestId("left-arrow");
  const rightArrow = getByTestId("right-arrow");

  // expect the left arrow to be missing, but the right button to be present.
  expect(leftArrow).toHaveClass("hidden");
  expect(rightArrow).not.toHaveClass("hidden");

  // move forward, expect both arrow to exist
  fireEvent.click(getByTestId("right-arrow"));
  // expect the left arrow to be missing, but the right button to be present.
  expect(leftArrow).not.toHaveClass("hidden");
  expect(rightArrow).not.toHaveClass("hidden");

  // move forward again, expect only the right arrow to be missing
  fireEvent.click(rightArrow);
  expect(leftArrow).not.toHaveClass("hidden");
  expect(rightArrow).toHaveClass("hidden");
});
