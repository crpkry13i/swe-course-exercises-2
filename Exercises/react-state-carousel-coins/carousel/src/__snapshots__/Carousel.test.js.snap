// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Carousel"
  >
    <h1>
      Shells from far away beaches.
    </h1>
    <div
      class="Carousel-main"
    >
      <i
        class="fas fa-chevron-circle-left fa-2x hidden"
        data-testid="left-arrow"
      />
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          Photo by Richard Pasquarella on Unsplash
        </h4>
        <img
          alt="Photo by Richard Pasquarella on Unsplash"
          class="Card-image"
          src="image1.jpg"
        />
        <small
          class="Card-small"
        >
          Image 1 of 3.
        </small>
      </div>
      <i
        class="fas fa-chevron-circle-right fa-2x "
        data-testid="right-arrow"
      />
    </div>
  </div>
</DocumentFragment>
`;
